applications/CompressiblePotentialFlowApplication/python_scripts/define_embedded_wake_process_2d.py

```diff
@@ -0,0 +1,59 @@
+import KratosMultiphysics
+import KratosMultiphysics.CompressiblePotentialFlowApplication as CPFApp
+import KratosMultiphysics.MeshingApplication as MeshingApplication
+from KratosMultiphysics.CompressiblePotentialFlowApplication.define_wake_process_2d import DefineWakeProcess2D
+import time
+import math
+
+
+def Factory(settings, Model):
+    if( not isinstance(settings,KratosMultiphysics.Parameters) ):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+
+    return DefineEmbeddedWakeProcess(Model, settings["Parameters"])
+
+
+class DefineEmbeddedWakeProcess(KratosMultiphysics.Process):
+    def __init__(self, Model, settings):
+        # Call the base Kratos process constructor
+        KratosMultiphysics.Process.__init__(self)
+
+        # Check default settings
+        default_settings = KratosMultiphysics.Parameters(r'''{
+            "model_part_name": "",
+            "epsilon": 1e-9
+        }''')
+        settings.ValidateAndAssignDefaults(default_settings)
+
+        self.main_model_part = Model[settings["model_part_name"].GetString()].GetRootModelPart()
+        self.wake_model_part=Model.CreateModelPart("wake")
+
+        self.epsilon = settings["epsilon"].GetDouble()
+
+    def ExecuteInitializeSolutionStep(self):
+        ini_time = time.time()
+
+        self._DefineWakeModelPart()
+        self._MoveAndRotateWake()
+        # Executing define wake process
+        CPFApp.DefineEmbeddedWakeProcess(self.main_model_part, self.wake_model_part).Execute()
+
+        KratosMultiphysics.Logger.PrintInfo('EmbeddedWake','Wake computation time: ',time.time()-ini_time)
+
+    def _DefineWakeModelPart(self):
+        ''' This function generates the modelpart of the wake. TODO: make end of the domain user-definable.
+        '''
+        self.wake_model_part.CreateNewNode(1, 0.0, 0.0, 0.0)
+        self.wake_model_part.CreateNewNode(2, 200.0, 0.0, 0.0)
+        self.wake_model_part.CreateNewElement("Element2D2N", 1, [1,2], KratosMultiphysics.Properties(0))
+
+    def _MoveAndRotateWake(self):
+        ''' This function moves and rotates the wake with the same parameters as the geometry.
+        '''
+        self.moving_parameters = KratosMultiphysics.Parameters()
+        self.moving_parameters.AddEmptyValue("origin")
+        self.moving_parameters["origin"].SetVector(self.main_model_part.ProcessInfo.GetValue(CPFApp.WAKE_ORIGIN))
+        self.moving_parameters.AddEmptyValue("rotation_angle")
+        angle=math.radians(-self.main_model_part.ProcessInfo.GetValue(CPFApp.ROTATION_ANGLE))
```

Comment:
I partly answered this [here, in the first answer](https://github.com/KratosMultiphysics/Kratos/pull/5261#issuecomment-512359379).

If im not mistaken, the angle of the wake in two dimensions does not affect the final solution. Here is an exmaple I just ran with a body fitted mesh with a wake angle of 30º,not matching the free stream velocity, which gives a cl=0.6018333878039077 

![image](https://user-images.githubusercontent.com/32136457/61395019-b8dc1c00-a8c4-11e9-83c8-f52fb42bcb6c.png)

Im taking advantage of this to just move the wake as I move the geometry (because its simpler).

The next thing to do would be to do it properly by separating the model part into parts, and defining the "inside-geometry wake" with its corresponding angle and the "outside wake" with its corresponding angle, which will be needed in 3D where the wake is fixed.

Reply:
I see, in that case it makes sense. The only thing is, do you think it would make sense to call the "inside-geometry wake" as camber line?
![Wing_profile_nomenclature svg](https://user-images.githubusercontent.com/28628414/61444233-46624f00-a94b-11e9-962b-9de96c350d28.png)

Reply:
Yes, absolutely!